conda_build/main_inspect.py

```diff
@@ -229,26 +230,27 @@ def test_installable(channel='defaults', verbose=True):
         print("######## Testing platform %s ########" % platform)
         channels = [channel] + get_default_urls()
         index = get_index(channel_urls=channels, prepend=False, platform=platform)
-        for package in sorted(index):
-            if channel != 'defaults':
-                # If we give channels at the command line, only look at
-                # packages from those channels (not defaults).
-                if index[package]['channel'] not in normalize_urls([channel], platform=platform):
-                    continue
-            name, version, build = package.rsplit('.tar.bz2', 1)[0].rsplit('-', 2)
```

Comment:
This is critical: we never want to depend on the package filename for this info. We need to extract it from the index instead.